### ISSUE TITLE
[Platform][k8s] Support publishing NODE_IP instead of POD_IP as the broadcast address 

### ISSUE THREAD

**Issue by mheer9** [yugabyte/yugabyte-db#10603]
Jira Link: [DB-1214](https://yugabyte.atlassian.net/browse/DB-1214)
### Description

We have a ticket opened by a customer, regarding their k8's helm chart. 2243
My understanding is in our Helm chart we use Pod IP while deploying the cluster. They want Node IP to be used instead, since that is what will be exposed to the applications.

So the change has to be done in Helm chart from our end, so that when ever they deploy it, NodeIP is used.
They have attached a zip of their helm chart files, there are a lot of changes, but we are only interested in the changes related to NODE_IP in file values.yaml. Updated the ticket with this info, as well as adding a link to a slack conversation that took place recently.

yb.zip is available on the lincoln server under /cases/2243.

**Comment by mheer9** [comment]
Please find the discussion Gautam started with Engg. team.

https://yugabyte.slack.com/archives/CC5C65CP7/p1634574228168300

Summary: Customer has been using NODE IP configuration for the whole of testing phase so they want to use the same in production as well.

Changes related to NODE_IP are as following

values.yaml

tserver:
externalBroadcast: true
serviceBroadcastAddress: "[$(NODE_IP)]"

master:
externalBroadcast: false

templates/service.yaml

containers:

   env:
    - name: NODE_IP
      valueFrom:
        fieldRef:
          fieldPath: status.hostIP
master additional flag:
- "--server_broadcast_addresses="

tserver additional flag:
- "--server_broadcast_addresses="

**Comment by mheer9** [comment]
Jason Vigil has also asked the question below

Does the Platform need to connect to master/tserver server_broadcast_addresses in order to perform universe setup / configuration tasks? If so, how will the platform determine the node IP address value to use?

@iSignal

**Comment by bhavin192** [comment]
With the changes merged into master, and being backported to 2.6, 2.8. After corresponding releases are out, this can be used by adding following YAML to each zone override config during provider creation in the platform.

```yaml
master:
  podAnnotations:
    prometheus.io/path: "/prometheus-metrics"
    # Any other required annotations

tserver:
  extraEnv:
  - name: NODE_IP
    valueFrom:
      fieldRef:
        fieldPath: status.hostIP
  serverBroadcastAddress: "$(NODE_IP)"
  affinity:
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
      - labelSelector:
          matchExpressions:
          - key: app
            operator: In
            values:
            - "yb-tserver"
        topologyKey: kubernetes.io/hostname
  podAnnotations:
    prometheus.io/path: "/prometheus-metrics"
    # Any other required annotations

# Required to make sure that the Kubernetes FQDNs are used for
# internal communication between the nodes, and node to node
# TLS certificates are validated correctly.

gflags:
  master:
    use_private_ip: cloud
  tserver:
    use_private_ip: cloud

serviceEndpoints:
  - name: "yb-master-ui"
    type: LoadBalancer
    app: "yb-master"
    ports:
      http-ui: "7000"

  - name: "yb-tserver-service"
    type: NodePort
    externalTrafficPolicy: "Local"
    app: "yb-tserver"
    ports:
      tcp-yql-port: "9042"
      tcp-yedis-port: "6379"
      tcp-ysql-port: "5433"
```

**Comment by streddy-yb** [comment]
Thanks @bhavin192. Hi @lizayugabyte - We need to document this as part of our public k8s documentation. I will file a separate docs issue for tracking.
